DFSNotifierResourcesDeployment/modules/configurationstore/keyVault.bicep

```diff
@@ -0,0 +1,105 @@
+param featureName string
+@allowed([
+  'westus2'
+  'southeastasia'
+])
+param deploymentLocation string
+@allowed([
+  'ppe'
+  'prod'
+  'perf'
+])
+param environment string
+param tagsObject object
+param accessPolicyObjectIds array
+param keyPermissionList array
+param secretPermissionList array
+param certificatePermissionList array
+
+param createMode string
+param shortLocation string = ''
+param serviceKeyPermissionList array = []
+param serviceSecretPermissionList array = []
+param serviceCertificatePermissionList array = []
+
+var envBasedName = environment != 'prod' ? 'dfsn-${featureName}-${empty(shortLocation) ? deploymentLocation : shortLocation}-${environment}' : 'dfsn-${featureName}-${empty(shortLocation) ? deploymentLocation : shortLocation}'
+var keyVaultName_var = length(envBasedName) > 24 ? 'dfsn-${featureName}-${environment}' : envBasedName
+var tenantId = subscription().tenantId
+
+resource keyVault 'Microsoft.KeyVault/vaults@2019-09-01' = {
+  name: keyVaultName_var
+  location: deploymentLocation
+  tags: tagsObject
+  properties: {
+    createMode: createMode // This needs to be set to 'default' first time with below commented access policies
+    enableSoftDelete: true
+    enabledForDeployment: true
+    enabledForTemplateDeployment: true
+    enabledForDiskEncryption: true
+    tenantId: tenantId
+    accessPolicies: [for object in accessPolicyObjectIds: createMode == 'default' ? object.type == 'system-principal' ? {
+      tenantId: tenantId
+      objectId: object.key
+      permissions: {
+        keys: keyPermissionList
+        secrets: secretPermissionList
+        certificates: certificatePermissionList
+      }
+    } : {
+      tenantId: tenantId
+      objectId: object.key
+      permissions: {
+        keys: serviceKeyPermissionList
+        secrets: serviceSecretPermissionList
+        certificates: serviceCertificatePermissionList
+      }
+    } : {}]
+    sku: {
+      name: 'standard'
+      family: 'A'
+    }
+  }
+}
+
+// accessPolicies: [for object in accessPolicyObjectIds: createMode == 'default' ? object.type == 'system-principal' ? {
```

Comment:
Are we removing the commented code ?

Reply:
it is required in case of key vault.
While redeploying same resource file won't work like other resources here, we need use commented code.